lib/core/versions/latest/OperationProcessor.ts

```diff
@@ -97,6 +97,11 @@ export default class OperationProcessor implements IOperationProcessor {
         }
 
         AnchoredOperation.applyPatchesToDidDocument(didDocument, operation.patches!);
+
+        // Update the document with the actual id .. for example, the patches may add some fields that may not
+        // have the 'id' fields set.
+        Document.updatePlaceholdersInDocumentWithDid(didDocument, didDocument.id);
```

Comment:
Design: I feel it is better to hide/perform this logic behind `applyPatchesToDidDocument()`, and to that point, I think a good place actually do this would be within the `applyPatchToDidDocument()` (note NOT `applyPatchesToDidDocument()`) directly under handling of `add-public-keys`.